Src/Support/Google.Apis.Auth.AspNetCore/GoogleAuthProvider.cs

```diff
@@ -81,19 +87,25 @@ internal class GoogleAuthProvider : IGoogleAuthProvider
                 try
                 {
                     var refreshResponse = await options.Backchannel.PostAsync(oidcConfig.TokenEndpoint, refreshContent, cancellationToken);
+                    refreshResponse.EnsureSuccessStatusCode();
                     var payload = JObject.Parse(await refreshResponse.Content.ReadAsStringAsync());
                     var refreshedAccessToken = payload.Value<string>("access_token");
                     var refreshedRefreshToken = payload.Value<string>("refresh_token");
                     var refreshedExpiresIn = payload.Value<string>("expires_in");
+                    var refreshedIdToken = payload.Value<string>("id_token");
                     auth.Properties.UpdateTokenValue(OpenIdConnectParameterNames.AccessToken, refreshedAccessToken);
                     if (!string.IsNullOrEmpty(refreshedRefreshToken))
                     {
                         auth.Properties.UpdateTokenValue(OpenIdConnectParameterNames.RefreshToken, refreshedRefreshToken);
                     }
-                    auth.Properties.IssuedUtc = _clock.UtcNow;
                     if (int.TryParse(refreshedExpiresIn, out int expiresInSeconds))
                     {
-                        auth.Properties.ExpiresUtc = _clock.UtcNow + TimeSpan.FromSeconds(expiresInSeconds);
+                        var refreshedExpiresAt = now + TimeSpan.FromSeconds(expiresInSeconds);
+                        auth.Properties.UpdateTokenValue("expires_at", refreshedExpiresAt.ToString("o"));
```

Comment:
It's a pity there's no constant for OpenIdConnectParameterNames here :( I don't suppose this is present somewhere else?

Reply:
Agreed. I cannot see it defined anywhere, and the code in OpenIdConnectHandler.cs referenced in the comment above also just has this as a string literal :(